app/src/main/java/de/xikolo/presenters/video/VideoPresenter.java

```diff
@@ -41,7 +49,24 @@
     public void onViewAttached(VideoView view) {
         super.onViewAttached(view);
 
-        getViewOrThrow().setupVideo(course, section, item, video);
+        if (subtitles == null) {
+            itemManager.requestSubtitlesWithCuesForVideo(videoId, new RequestJobCallback() {
```

Comment:
So why loading them here at all?

Reply:
And what about positioning and styling?

Reply:
The `vttUrl` is saved in the `SubtitleTrack` model. They need to be loaded, because we cannot be sure they are already in the database, unlike the video, course, section and item which have been loaded before.

Reply:
Styling and positioning is completely done by ExoPlayer according to the User's Android accessibility settings for subtitles. I plan to incorporate a button in the upcoming settings UI leading the user to those settings.

Reply:
On Android 8.0 with default subtitle settings it looks like this:
![device-2018-08-29-165119](https://user-images.githubusercontent.com/26904189/44795803-d9678e80-abab-11e8-8a77-a71224c8fd8f.png)

Reply:
The video model already embeds them: https://dev.xikolo.de/api-docs/#endpoint-videos

Reply:
And schema documentation is coming very soon: https://dev.xikolo.de/gitlab/xikolo/xikolo-web/merge_requests/931/diffs#bf8df63b460ca22fcd7622eec300119c16231a8b_115_96